public/js/dom.js

```diff
@@ -0,0 +1,16 @@
+let btn =document.querySelector('button');
+let input = document.querySelector('input')
+let jobsContainer=document.querySelector('.static-jobs')
+btn.addEventListener('click',(e)=>{
+    jobsContainer.textContent=''
+    let load = createElement('div', 'loader');
+    jobsContainer.appendChild(load);
+    fetchNode('/search',{q:input.value}).then((data)=>{
+        if(data.status==500){
+            window.location.href='../errors/500.html'
+        }
+        else{
+            data.json().then(data=>generateJopCards(data,jobsContainer))
```

Comment:
nested promises :grimacing: 
it is better to use chaining by returning the promise and resolving it with .then instead of nesting your promises(this way you might end up with something like the callback hell but this time with promises)

```js
    fetch(url).then((res) => {
    ...
    ...
    return res.json()
    }.then(data => ....)
```